### DISCUSSION TITLE
`fetch` not defined when building 🤔

### DISCUSSION THREAD

**Discussion by YuukanOO** [ElMassimo/iles#41] [Q&A] [answered]
Hi there! Awesome work!

I'm currently testing iles out to see if it would fit my needs and I'm impressed. I just got one question about data fetching. I attempted to do something like this (as if I was getting data from an headless CMS):

```vue
<script setup lang="ts">
const todo = await fetch("https://jsonplaceholder.typicode.com/todos/1").then(
  (r) => r.json()
);
</script>

<template>
  <h1>{{ todo.title }}</h1>
</template>
```

And while it works as expected in dev mode, when running `npm run build`, I got a `ReferenceError: fetch is not defined` error. Did I miss something?

**Comment by YuukanOO** [comment, marked as answer]
Ok nevermind, I was just dumb. I forgot that node does not come with `fetch` so it can't work without relying on another module such as **node-fetch** or **isomorphic-fetch**.

**Reply by ElMassimo** [reply]
I'll likely provide `$fetch` (with auto-import like the Iles composables) using ohmyfetch out of the box, since this is a very common use case.

Thanks for sharing!
